homeassistant/components/homekit_controller/config_flow.py

```diff
@@ -154,33 +154,16 @@ async def async_step_unignore(self, user_input):
         if self.controller is None:
             await self._async_setup_controller()
 
-        devices = await self.controller.discover_ip(max_seconds=5)
-        for device in devices:
-            if normalize_hkid(device.device_id) != unique_id:
-                continue
-            record = device.info
-            return await self.async_step_zeroconf(
-                zeroconf.ZeroconfServiceInfo(
-                    host=record["address"],
-                    port=record["port"],
-                    hostname=record["name"],
-                    type="_hap._tcp.local.",
-                    name=record["name"],
-                    properties={
-                        "md": record["md"],
-                        "pv": record["pv"],
-                        zeroconf.ATTR_PROPERTIES_ID: unique_id,
-                        "c#": record["c#"],
-                        "s#": record["s#"],
-                        "ff": record["ff"],
-                        "ci": record["ci"],
-                        "sf": record["sf"],
-                        "sh": "",
-                    },
-                )
-            )
+        try:
+            device = await self.controller.find_ip_by_device_id(unique_id)
+        except aiohomekit.AccessoryNotFoundError:
+            return self.async_abort(reason="accessory_not_found_error")
+
+        self.name = device.info["name"].replace("._hap._tcp.local.", "")
```

Comment:
I'll be adding typed properties for these soon, in the meantime these replicate what currently happens.